### PR TITLE
restapi_vxlan_ecmp_test on mlnx 4600 and 8000 T1

### PR DESCRIPTION
**Pull request by siqbal1986**
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/sonic-net/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
Fixes # (issue)
Enabled the Test on 4600C T1.
### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)


### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [X] 202311
- [X] 202405

### Approach
#### What is the motivation for this PR?
Better test coverage.
#### How did you do it?
Enabled the test.
#### How did you verify/test it?
Ran the test naually on 4600
```
$ ./run_tests.sh -n  vms18-t1-msn4600c-acs-1  -i ../ansible/str2,../ansible/veos -r -u -c restapi/test_restapi_vxlan_ecmp.py
=== Running tests in groups ===
Running: python3 -m pytest restapi/test_restapi_vxlan_ecmp.py --inventory ../ansible/str2,../ansible/veos --host-pattern str2-msn4600c-acs-01 --testbed vms18-t1-msn4600c-acs-1 --testbed_file /var/src/sonic-mgmt-int/ansible/testbed.yaml --log-cli-level warning --log-file-level debug --kube_master unset --showlocals --assert plain --show-capture no -rav --allow_recover --ignore=ptftests --ignore=acstests --ignore=saitests --ignore=scripts --ignore=k8s --ignore=sai_qualify --junit-xml=logs/tr.xml --log-file=logs/test.log
==================================================================================================================================== test session starts =====================================================================================================================================
platform linux -- Python 3.8.10, pytest-7.4.0, pluggy-1.5.0
ansible: 2.13.13
rootdir: /var/src/sonic-mgmt-int/tests
configfile: pytest.ini
plugins: metadata-3.1.1, allure-pytest-2.8.22, forked-1.6.0, repeat-0.9.3, html-4.1.1, ansible-4.0.0, xdist-1.28.0
collecting ...
------------------------------------------------------------------------------------------------------------------------------------ live log collection -------------------------------------------------------------------------------------------------------------------------------------
23:16:45 testbed.get_testbed_type                 L0261 WARNING| Unsupported testbed type - point-to-point-256
/usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
/usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
/usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
/usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
/usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
collecting 0 items                                                                                                                                                                                                                                                                           23:17:11 testbed.get_testbed_type                 L0261 WARNING| Unsupported testbed type - point-to-point-256
collected 1 item

restapi/test_restapi_vxlan_ecmp.py::test_vxlan_ecmp_multirequest PASSED                                                                                                                                                                                                                [100%]

```
#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

### CODE REVIEW

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 1]
```diff
@@ -1505,7 +1505,7 @@ restapi/test_restapi_vxlan_ecmp.py:
   skip:
     reason: "Only supported on cisco 8102 T1"
     conditions:
-      - "not (asic_type in ['cisco-8000'] and 't1' in topo_type)"
+      - "not (platform in ['x86_64-8102_64h_o-r0', 'x86_64-8101_32fh_o-r0', 'x86_64-mlnx_msn4600c-r0']  and 't1' in topo_type)"
```
Comment (bingwang-ms) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:1508]: Please double check the skip statement. `not t1 in topo_type` means the test will be skipped in Non-T1 topo, but the code specify the test to be run on T1.

Comment (siqbal1986) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:1508]: It should be run on 8102, 8101, 4600 T1s. I am not sure I understand the comment here.

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 2]
```diff
@@ -1505,7 +1505,7 @@ restapi/test_restapi_vxlan_ecmp.py:
   skip:
     reason: "Only supported on cisco 8102 T1"
```
Comment (bingwang-ms) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:1506]: Please update the reason

**Review by bingwang-ms — APPROVED** [review]

**Comment by yutongzhang-microsoft** [comment]
@siqbal1986 , In this PR, you are going to onboarding this script on T1 topology, but I don't think we should change the pytest topology mark from `t0` to `t1`, as this script is also fit for T1 topology. Instead, we should **add**  `t1` into pytest topology mark

**Pull request merged**